Remove SAML connector and tests

camscale opened the pull request:
Remove the SAML connector from the OSS repository. It has been migrated
to the enterprise repository.

Issue: https://github.com/gravitational/teleport.e/issues/525

---

NOTE: This PR should be reviewed in conjunction with https://github.com/gravitational/teleport.e/pull/597
which adds the SAML connector to the enterprise repo.

camscale committed:
auth: Remove SAML connector code

Remove the SAML connector code as it now lives in the enterprise
repository. All that is left behind is a small forwarding stub that
returns a NotImplemented error if a SAMLService is not plugged into the
auth.Server.

camscale committed:
auth: Remove remaining SAML tests

Remove the SAML tests that were not in `saml_test.go`. These tests have
already been moved to the enterprise repository along with the SAML
connector.

camscale committed:
Update enterprise submodule

ravicious commented on lib/auth/fuzz_test.go:
```diff
@@ -17,24 +17,11 @@ limitations under the License.
 	"github.com/stretchr/testify/require"
 )
 
-func FuzzParseSAMLInResponseTo(f *testing.F) {
```
I'm not familiar with this part of the codebase, but are we okay with removing this fuzz test completely? From what I see, it wasn't moved to teleport.e.

smallinsky replied:
The reference to this test can be also found in https://github.com/gravitational/teleport/blob/fd4097b93190b6580df793932fe6946aa343b300/fuzz/oss-fuzz-build.sh#L30 

So probably we need to remove it from there.

camscale replied:
Ah yes. I over-zealously removed this (the always-failing CIFuzz annoys me no end), and forgot to see what I needed to do to move it to the new repo. I'll get onto that.

camscale replied:
I've added the fuzz test to the enterprise PR. As yet there is nothing to run it as there is no CI job there to do fuzz testing. I can't figure out how the `fuzz/oss-fuzz-build.sh` script is meant to work - I think I'm missing the context in which it runs - there look to be undefined functions and variables, so I suspect OSSFuzz provides this context and corpora. A similar setup will need to be done on the enterprise repo.

@reedloden Is there any feedback you can add?

reedloden replied:
OSS-Fuzz runs `fuzz/oss-fuzz-build.sh` from the config in https://github.com/google/oss-fuzz/tree/master/projects/teleport.

OSS-Fuzz only works on open source projects. We can use https://fuzzbuzz.io/ for `teleport.e`, but it's not configured yet. As long as you ensure Go-native fuzzing is set up (by way of the fuzz test existing), when we do get Fuzzbuzz working, it will Just Work(TM).

smallinsky commented on lib/auth/auth_with_roles.go:
```diff
@@ -2821,9 +2821,6 @@ func (a *ServerWithRoles) UpsertSAMLConnector(ctx context.Context, connector typ
 	if err := a.authConnectorAction(apidefaults.Namespace, types.KindSAML, types.VerbUpdate); err != nil {
 		return trace.Wrap(err)
 	}
-	if !modules.GetModules().Features().SAML {
```
Wonder if there is a case when a client can have a Enterprise product without SAML feature ?

camscale replied:
Not currently as far as I'm aware. This check is still performed but in a different way. If the enterprise startup does not register the SAML connector, you'll still get this error message from the code in `lib/auth/saml.go` in the OSS repo. If we do get a scenario where we can have an enterprise product without SAML, we can revise the error message then since I don't know how to word it otherwise right now.

camscale committed:
fuzz: Remove run of FuzzParseSAMLInResponseTo

This fuzz test has moved to the enterprise repository, so remove the run
from here.

camscale commented:
@ravicious @smallinsky  This PR has a failing test for the operator that I did not notice before as it is not run from `make test-go` on the command line. I've raised https://github.com/gravitational/teleport/pull/18634 to keep some auth/SAML methods in `auth.Server`. Once I get that merged, I'll rebase this and the other PR on top of that. I think some of the tests that moved across will end up staying now as those related to Upsert/Delete of the connector will be fully implemented in this OSS repo.

camscale committed:
auth: Add back TestSAMLValidation

Add `TestSAMLValidation` back to `grpcserver_test.go` as it does not do
actual SAML auth, just creating a connector. That is fully supported in
OSS - only the auth path required enterprise.

camscale committed:
auth: Add missing "fmt" import

Add a "fmt" import that got lost in a rebase.

camscale committed:
Merge branch 'master' into camh/remove-saml
